File: Python/Longest Palindromic Substring.py
class Solution:
    def longestPalindrome(self ,arr):
        res=""
        reslen=0
        for i in range(len(arr)):
            l=i
            r=i
            while l>=0 and r<len(arr) and arr[l]==arr[r]:
                if r-l+1>reslen:
                    res=arr[l:r+1]
                    reslen=r-l+1
                l-=1
                r+=1
            l=i
            r=i+1     
            while l>=0 and r<len(arr) and arr[l]==arr[r]:
                if r-l+1>reslen:
                    res=arr[l:r+1]
                    reslen=r-l+1
                l-=1
                r+=1
        return res

File: Java/Longest Palindromic Substring.java
class Solution {
    public String longestPalindrome(String s) {
        String s1="";
        int n=s.length();
        int L;
        for(int i=0 ; i < s.length();i++){
            L=i;

            if (s1.length()<expand(L,L,n,s).length()){
                s1=expand(L,L,n,s);

            }
             if (s1.length()<expand(L,L+1,n,s).length()){
                s1=expand(L,L+1,n,s);

            }
        }

        return s1;
        
    }

    public String expand(int L , int R , int n, String s){
        while (L>=0 && R<n){
            if(s.charAt(L)!=s.charAt(R)){
                break;
            }
            L--;
            R++;
        }
        return s.substring(L+1,R);


    }
}
    
